docker/apm-server/managed/main.go

```diff
@@ -0,0 +1,245 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+
+	"github.com/pkg/errors"
+)
+
+func main() {
+	setup()
+	// create a reverse proxy to the APM Server running via Elastic Agent,
+	// headers are not rewritten, as not considered important right now.
+	target, _ := url.Parse("http://elastic-agent:8200")
+	http.Handle("/", httputil.NewSingleHostReverseProxy(target))
+	if err := http.ListenAndServe(":8200", nil); err != nil {
+		log.Fatal(err)
+		os.Exit(-1)
+	}
+}
+
+func setup() {
+	client := newKibanaClient()
+	if err := cleanup(client); err != nil {
```

Comment:
What if this happens on container restart?  is there a way to check if we're already attached the default policy and do nothing in that case?

Reply:
Made some changes and now check if the existing policy fits the expectations - makes the code slightly more complicated, but has a bit faster startup time.